app/services/augment_marc_record_service.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# Service class for augmenting source MARC records with POD information
+class AugmentMarcRecordService
+  attr_reader :steps, :organization
+
+  DEFAULT_STEPS = %i[
+    append_pod_provenance_information
+    append_normalized_item_information
+  ].freeze
+
+  def initialize(steps: nil, organization: nil)
+    @steps = steps || DEFAULT_STEPS
+    @organization = organization || Organization.new
+  end
+
+  # @param [MARC::Record] record to modify
+  def execute(source_record)
+    steps.each_with_object(duped_record(source_record)) do |step, record|
+      send(step, record)
+    end
+  end
+
+  private
+
+  # Append a MARC 900$b containing the organization code of the organization that
+  # provided the record to POD.
+  # @param [MARC::Record] record to modify
+  def append_pod_provenance_information(record)
+    return if organization.code.blank?
+
+    record.append(MARC::DataField.new('900', nil, nil, pod_source_metadata, ['b', organization.code]))
+  end
+
+  # Perform organization-specific/specified normalization steps
+  # and add a $8 with a reference back to the source field (given as linking number = tag, sequence number = index)
+  # @param [MARC::Record] record to modify
+  def append_normalized_item_information(record)
+    return if organization.normalization_steps.blank?
+
+    organization.normalization_steps.each_value do |step|
+      map_normalized_item_field(record, step)
+    end
+  end
+
+  def map_normalized_item_field(record, step)
+    record.fields(step['source_tag']).each.with_index do |field, index|
+      subfield_data = map_subfield_data(field, step['subfields'])
+      field_link = ['8', "#{field.tag}.#{index}\\p"]
```

Comment:
I'm not sure if it matters much, but the typical formatting of $8 would be something like `999-01`, `999-02`, etc. 

I suppose we can't count on there only being 100 fields (for the padded zeros) but we might want to consider using a hyphen to concatenate the tag and the index to be consistent w/ other MARC fields.

Reply:
I'm not familiar w/ the `/p` what does that signify?

Reply:
Hm. I made that linking field up based on the the MARC21 spec that says:
> $8 [linking number].[sequence number]\[field link type]

and 
> p - Metadata provenance  Used in a record to link a field with another field containing information about provenance of the metadata recorded in the linked field.
(which.. is maybe technically wrong, but seemed more appropriate than other choices)

Reply:
Ah, I'm thinking about $6 actually, which uses `-` instead of `.` because of course it does.